controller/src/main/java/io/pravega/controller/store/stream/tables/TableHelper.java

```diff
@@ -184,6 +185,56 @@ public static StreamTruncationRecord computeTruncationRecord(final byte[] indexT
                 previousTruncationRecord.getDeletedSegments(), ImmutableSet.copyOf(toDelete));
     }
 
+    /**
+     * A method to compute size of stream in bytes from start till given stream cut.
+     * Note: this computed size is absolute size and even if the stream has been truncated, this size is computed for the
+     * entire amount of data that was written into the stream.
+     *
+     * @param indexTable index table for the stream
+     * @param historyTable history table for the stream
+     * @param segmentTable segment table for the stream
+     * @param streamCut stream cut to compute size till
+     * @param sealedSegmentsRecord record for all the sealed segments for the given stream.
+     * @return size (in bytes) of stream till the given stream cut.
+     */
+    public static long computeSizeTill(final byte[] indexTable, final byte[] historyTable, final byte[] segmentTable,
```

Comment:
Isn't `getCutSize` or `getStreamCutSize` better for the method name?

Reply:
a cut in itself doesnot have a size..  its a point.. we are computing size of the stream till the given cut.. hence the name computeSizeTill (StreamCut).

Reply:
> a cut in itself doesnot have a size.. its a point.. 

I don't use the same definition. I use the definition of a cut as proposed by Babaoglu and Marzullo:

_A cut of a distributed computation is a subset of its global history   and contains an initial prefix of each of the local histories._

http://www.cs.unibo.it/babaoglu/papers/pdf/global-states.pdf
(see page 7)

As as subset, it is amenable to having a size.

> we are computing size of the stream till the given cut.. hence the name computeSizeTill (StreamCut).

There are 5 parameters so `...Till(5 parameters)` is not super clear. it is fine if you don't like my suggestions, but please find a better name.

Reply:
That's true, a cut can also refer to the beginning of a range in the stream. I still dislike the method name, though. A couple of reasons:

- We don't typically use `compute` for the action, `get` is more commonly used in our code.
- `till` is vague, it doesn't explicitly says what it is referring to.

Reply:
changed the name to `getSizeTillStreamCut`.